app.rb

```diff
@@ -3,6 +3,7 @@
 require_relative 'rental'
 require_relative 'student'
 require_relative 'teacher'
+require_relative 'user_interface'
 
 class App
   attr_accessor :people, :books, :running, :rentals
```

Comment:
- [ ] Great work refactoring your code. However, there're few things to look into. I am going to leave all the comments on changes you're going to make. Reason for this is that I am unble to highllight those areas to make comment on. But I will do my best to screenshot those areas from my editor and upload them here. 

![create-method](https://github.com/ulises2607/School-Library/assets/26597432/e4d38e7c-3343-4196-9b09-4f59c63c587a)

![create-person](https://github.com/ulises2607/School-Library/assets/26597432/a5093151-6706-4261-99eb-dc80eb3878c7)

![list-method](https://github.com/ulises2607/School-Library/assets/26597432/e25a6021-6cc0-4c49-ab6a-169947b41edd)

- [ ] First and foremost, all requests for input and input validation should've been placed in the classes/methods whose responsibilities are to request or validate input. For proper implementation of single responsibility rule, you need to place  those methods in a file say `list.rb` and import them to your `app.rb` file to execute. Your  'list.rb' and  `app.rb` files should look something similar to this. :point_down:

``` 
   list.rb

   class List
  def books(list)
    puts 'No books' if list.empty?
    list.each do |book|
      puts "Title: \"#{book.title}\", Author: #{book.author}"
    end
    puts 'Press Enter to continue!'
  end

  def people(list)
    puts 'No people' if list.empty?
    list.each do |person|
      puts "[#{person.class}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
    end
    puts 'Press Enter to continue!'
  end

  def rentals(list)
    list.each do |person|
      puts "- [#{person.class}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
    end
    print 'ID of person: '
    person_id = gets.chomp.to_i
    person_obj = list.find { |person| person.id == person_id }
    puts 'Rentals:'
    person_obj.rentals.each do |rental|
      puts "Date: #{rental.date}, Book: \"#{rental.book.title}\" by #{rental.book.author}"
    end
    puts 'Press Enter to continue!'
  end
end

app.rb

require './book'
require './rental'
require './create'
require './list'

class App
  attr_reader :options

  def initialize(books_list = [], people_list = [])
    @books_list = books_list
    @people_list = people_list
    @list = List.new
  end

  def list_books
    @list.books(@books_list)
    gets.chomp
  end

  def list_people
    @list.people(@people_list)
    gets.chomp
  end

  def create_person
    @people_list << @create.person
    gets.chomp
  end

  def create_book
    @books_list << @create.book
    gets.chomp
  end

  def create_rental
    @create.rental(@books_list, @people_list)
    gets.chomp
  end

  def list_rentals
    @list.rentals(@people_list)
    gets.chomp
  end
end

```

- Kindly do more of refactoring so that your code could follow proper solid principle.